src/hrflow_connectors/connectors/talentsoft/warehouse.py

```diff
@@ -277,13 +378,71 @@ def read_profiles(
         params["offset"] += LIMIT
 
 
+def write_profiles(
+    adapter: LoggerAdapter,
+    parameters: WriteProfileParameters,
+    profiles: t.Iterable[t.Dict],
+) -> t.List[t.Dict]:
+    """Write profiles into TalentSoft"""
+    failed_profiles = []
+    adapter.info("Requesting Authentication Token from TS")
+    token = get_talentsoft_auth_token(
+        client_url=parameters.client_url_front,
+        client_id=parameters.client_id_front,
+        client_secret=parameters.client_secret_front,
+        front_or_back="front",
+    )
+    adapter.info("Authentication with TS API Endpoint finished")
+    for profile in profiles:
+        attachment = profile.pop("attachment", None)
+        if not attachment:
+            adapter.error("No attachment found for profile={}".format(profile))
+            failed_profiles.append(profile)
+            continue
+        try:
+            cv_content = get_cv_content(attachment)
+        except Exception as e:
+            adapter.error("Failed to get cv content with response={}".format(e))
+            failed_profiles.append(profile)
+            continue
+        filename = attachment["original_file_name"]
+        mime_type = get_mime_type_with_mimetypes(filename)
+        files = [
+            (
+                "cv_file_id",
+                (filename, cv_content, mime_type),
+            )
+        ]
+        if parameters.job_reference:
+            profile["application"]["offerReference"] = parameters.job_reference
+        profile_ts = dict(applicantApplication=json.dumps(profile))
+        profile_ts = decode_json(profile_ts)
```

Comment:
Ici je comprends pas à quoi sert la fonction `decode_json` et ce qu'elle utilise en dessous `decode_unicode`. Pourquoi tu as besoin de ce traitement ? 


Aussi sauf erreur de ma part la fonction `decode_json` est utilisé juste ici. Et vu que tu `profile_ts = dict(applicantApplication=json.dumps(profile))` tu te retrouve avec `profile_ts: {applicantApplication: str}` donc ta fonction fait te renvoit juste `{applicantApplication : decode_unicode(json.dumps(profile))}` 

Si tu as vraiment besoin de l'effet de `decode_unicode` enleve la fonction `decode_json` et fait juste ce dont tu as besoin 
```
profile_ts = dict(applicantApplication=decode_unicode(json.dumps(profile)))
```

Mais explique moi avant pourquoi tu as besoin de `decode_unicode`

Reply:
@the-forest-tree lors de mes tests j'ai constaté qu'il y a parfoit des caractère spéciale dans le profile dont leur envoit cause une erreur côté talentsoft.
Donc l'idée du code c'est de lire tout l'objet et transformé les caractère spéciale.
c'est une sorte de fonction recurrente qui s'auto appele.
Merci

Reply:
Ok donc je dois ne pas faire json.dumps en première. c'est ça?

Reply:
Donne moi un example de caractère spécial qui marche pas

Reply:
Je retest et je répond ici

Reply:
J'ai effectué les modifications evoqué.
Merci